<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>siMEM by neellab</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-dark.css">
    <script src="javascripts/scale.fix.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <align="center">        
        <h1>siMEM algorithm for the prediction of differential essentiality from pooled functional/phenotypic (shRNA, CRISPR/Cas9) screens</h1>
        <!-- <p>-->

          Data for &quot;The Functional Genomic Landscape  of Human Breast Cancer Drivers, Vulnerabilities, and Resistance&quot;
        </align>
        <!--</p>-->
      </header>
      <section>
        <h3>
<a id="manuscript" class="anchor" href="#manuscript" aria-hidden="true"><span class="octicon octicon-link"></span></a>Manuscript</h3>

<p><strong><em>"Functional Genomic Landscape  of Human Breast Cancer Drivers,  Vulnerabilities, and Resistance"</em></strong></p>

<p>Richard Marcotte*, Azin Sayad*, Kevin R. Brown, Felix Sanchez-Garcia, Jüri Reimand, Maliha Haider, Carl Virtanen, James E. Bradner, Gary D. Bader, Gordon B. Mills, Dana Pe’er, Jason Moffat and Benjamin G. Neel</p>

<p>(* Equal Contributions)</p>

<p>[<strong><a href="http://www.cell.com/cell/home">link</a></strong>] [<strong><a href="">Pubmed</a></strong>] [<strong><a href="">PDF</a></strong>] [<strong><a href="http://neellab.github.io/bfg/">ACCOMPANYING DATA WEBSITE</a></strong>]</p>

<br>
<br>
<h3>
<a id="links-to-data-on-geo" class="anchor" href="#links-to-data-on-geo" aria-hidden="true"><span class="octicon octicon-link"></span></a>Links to Data on GEO</h3>

<p>shRNA screens (<strong><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74702">GSE74702</a></strong>)</p>

<p>RNAseq (<strong><a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE73526">GSE73526</a></strong>)</p>


<br>
<br>
<h3>
<a id="functional-genomics-and-proteomics-data-files" class="anchor" href="#functional-genomics-and-proteomics-data-files" aria-hidden="true"><span class="octicon octicon-link"></span></a>Functional, Genomics and Proteomics Data Files</h3>

<p><strong>shRNA screens</strong></p>

<ul>
<li>
<strong><a href="https://www.dropbox.com/s/a16l4thfeu9u206/breast_screens.eset?dl=1">Normalized ExpressionSet</a></strong>: File used in manuscript analyses. RData format file containing a <code>Bioconductor ExpressionSet</code> object combining

<ul>
<li>Loaded into R using the <code>load()</code> function.</li>
<li>Per-sample (cell line + time-point + replicate), per-hairpin log2 measurements matrix. Accessed using the <code>exprs()</code> R function.</li>
<li>Per-hairpin, per-gene annotations data frame, with the same number of rows as the measurement matrix. Accessed using the <code>fData()</code> R function.</li>
<li>Per sample annotation data frame, including information such as cell line and time-point. This number of rows in this data frame matches the number of columns in the measurements matrix. Accessed using the <code>pData()</code> R function.</li>
</ul>
</li>
</ul>

<p><strong>RNASeq</strong></p>

<ul>
<li><strong><a href="http://RNASEQ">Per-gene log2 FPKM</a></strong></li>
<li>
<strong><a href="https://github.com/neellab/bfg/blob/gh-pages/data/rnaseq/breast_rnaseq_qn.txt.zip?raw=true">Per-gene normalized log2 FPKM</a></strong>: As above, but also Quantile-Normalized, and a floor value set at log2(0.1) FPKM. This version was used in the manuscript analyses.</li>
</ul>

<p><strong>CNA</strong></p>

<ul>
<li><strong><a href="https://github.com/neellab/bfg/blob/gh-pages/data/cna/breast_cna_lrr.txt.zip?raw=true">Per-gene Log-R Ratios</a></strong>:
CBS-segmented SNP array Log R-Ratios, mapped to genes.
</li>
</ul>

<p><strong>miRNA</strong></p>

<ul>
<li><strong><a href="">miRNA</a></strong></li>
</ul>

<p><strong>RPPA</strong></p>

<ul>
<li><strong><a href="https://github.com/neellab/bfg/blob/gh-pages/data/rppa/breast_rppa.xlsx?raw=true">RPPA</a></strong></li>
</ul>

<br>
<br>
<h3>
<a id="annotation-files" class="anchor" href="#annotation-files" aria-hidden="true"><span class="octicon octicon-link"></span></a>Annotation Files</h3>

<p><strong><a href="https://github.com/neellab/bfg/blob/gh-pages/data/annotations/cell_line_subtypes.txt.zip?raw=true">Cell line subtypes</a></strong>: sub-typing of each cell line according to</p>

<ul>
<li>Receptor expression and status (AR, ERBB2, ESR1, PGR).</li>
<li>Three-receptor classification (ER, HER2, TNBC).</li>
<li>Curtis Integrative Subtypes (IntClust 1 through 10).</li>
<li>Intrinsic classification (Luminal A, Luminal B, Her2, Basal, Normal-like) with Claudin Low status.</li>
<li>Lehmann TNBC subtypes (Basal Like 1, Basal Like 2, Immunomodulatory, Luminal Androgen Receptor, Mesenchymal, Mesenchymal Stem-Like).</li>
<li>Neve classification (luminal, basal A, basal B) with luminal cell lines further classified as her2 based on high ERBB2 expression status.</li>
<li>Unsupervised clustering of RNASeq, RPPA and shRNA screen data.</li>
</ul>

<p><strong><a href="https://github.com/neellab/bfg/blob/gh-pages/data/annotations/hairpin_annotations.txt.zip?raw=true">updated shRNA annotations</a></strong>:
Update to Entrez gene ids and symbols, to account for changed symbols, deprecated Entrez ids and the like. Approximately 300 gene ids from the original TRC II annotations no longer exist, leading to a slightly reduced overall gene id and shRNA count.
</p>
      </section>
    </div>
    <footer>
      <p>Project maintained by <a href="https://github.com/neellab">neellab</a> and<br>Azin Sayad (asayad@gmail.com)</p>
      <p>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></p>
    </footer>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
    
  </body>
</html>
